Sources/FirebaseFirestore/DocumentSnapshot+Swift.swift

```diff
@@ -9,9 +9,37 @@ import CxxShim
 import Foundation
 
 public typealias DocumentSnapshot = firebase.firestore.DocumentSnapshot
+public typealias ServerTimestampBehavior = firebase.firestore.DocumentSnapshot.ServerTimestampBehavior
+public typealias MapFieldValue = firebase.firestore.MapFieldValue
+
+public typealias GeoPoint = firebase.firestore.GeoPoint
+public typealias FieldValue = firebase.firestore.FieldValue
+
+extension DocumentSnapshot {
+  public var reference: DocumentReference {
+    swift_firebase.swift_cxx_shims.firebase.firestore.snapshot_reference(self)
+  }
+
+  public func data(with behavior: ServerTimestampBehavior) -> [String: Any]? {
+    let data = swift_firebase.swift_cxx_shims.firebase.firestore.snapshot_get_data(self, behavior)
+
+    return FirestoreDataConverter.value(workaround: data)
+  }
+}
+
+public extension DocumentSnapshot {
+    func data<T: Decodable>(as type: T.Type,
+                            with serverTimestampBehavior: ServerTimestampBehavior = .none,
+                            decoder: Firestore.Decoder =  .init()) throws -> T {
+    let value: Any = data(with: serverTimestampBehavior) ?? NSNull()
+    let result = try decoder.decode(T.self, from: value, in: reference)
```

Comment:
I simply find it easier to debug when using a debugger or when forced to use print debugging. Since debugging is somewhat hit or miss this felt easier as I was building. Is there a real performance trade off here?